x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -402,7 +407,7 @@ func createMetricDataQuery(metric cloudwatch.Metric, instanceID string, index in
 	for _, dim := range metricDims {
 		if *dim.Name == "InstanceId" && *dim.Value == instanceID {
 			metricName := *metric.MetricName
-			label := instanceID + labelSeparator + metricName + labelSeparator + statistic
+			label := constructLabel(instanceID, metricName, statistic)
```

Comment:
Nice application of these structures.

Reply:
Thanks for the suggestion so I can get rid of the labelSeparator and nasty parsing...